clippy_utils/src/lib.rs

```diff
@@ -1491,6 +1492,24 @@ pub fn is_else_clause(tcx: TyCtxt<'_>, expr: &Expr<'_>) -> bool {
     }
 }
 
+/// Checks whether the given `Range` is equivalent to a `RangeFull`.
+/// Inclusive ranges are not considered because they already constitute a lint.
+pub fn is_range_full(cx: &LateContext<'_>, container: &Expr<'_>, range: Range<'_>) -> bool {
+    range.start.map_or(true, |e| is_integer_const(cx, e, 0))
```

Comment:
Since you made this a `clippy_utils` function, we should take care to either document any possible shortcomings or rule them out. For example, I think the method will fail on a `Range<i8>`.

Reply:
Thx for pointing that out! It didn't cross my mind...

Unfortunately I don't see a way to make it work for signed integers because right now the test on the lower bound is performed with `is_integer_const`, which takes a `u128` as a value to check for, so I can't check for `i8::MIN`.

Do you have some way that that could be accomplished in mind? Otherwise, I will document the function more precisely.

I have to admit that the applicability of this function is a bit awkward, for example I don't think it works on `..usize::MAX` either. It was made specifically to detect `RangeFull`s or equivalents used to index `Vec`s and `VecDeque`s, but the way it's written it also works in a few other cases. Maybe I can tweak the implementation to only check for `RangeTo`, `RangeFrom` and `Range` if the range is the argument of a method call of a `Vec` or `VecDeque`, what do you think?

Reply:
I guess the easiest way is to rename to `is_usize_range_full` and document the restricted applicability of the method. Otherwise you can get the type (via `expr_ty`), then the generic, then match on that to figure out the bounds.

Reply:
It might even be better to restrict the check to ranges passed to `drain` for now (apart from `.. ` which is the easy case). That makes it less flexible but at the same time it does prevent any misuse.

Reply:
Thx for the pointer, I will try to get the generic case working and if I really can't I will fall back to the `usize` solution with added documentation.

Reply:
Cool! Feel free to ping us here or on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy) if you need further assistance.

Reply:
It took a fair amount of work but I think I got it! At least it works as expected in my use case, and I added a couple of tests where I use `usize::MIN` as a lower bound to make sure that works as well (it already worked before with the hard-coded 0 but I thought it would be nice to make sure that it works).

I haven't added tests for the case where signed integers are used because this lint only applies in a method call to `drain` which only takes `usize` ranges. Do you need me to add unit tests for the new `is_range_full` function before accepting this PR? If so, where are such tests placed in the repo?

Note that I had to significantly alter the function signature of `is_range_full` to make it more generic. For instance, I made the container optional so that this function can be called on any kind of range.